src/Authentication/AuthenticationManager.php

```diff
@@ -26,48 +37,50 @@ class RestfulAuthenticationManager extends \ArrayObject {
   protected $isOptional = FALSE;
 
   /**
-   * Set the authentications' "optional" flag.
-   *
-   * @param boolean $is_optional
-   *   Determines if the authentication is optional.
+   * @param AuthenticationPluginManager $manager
+   *   The authentication plugin manager.
+   */
+  public function __construct(AuthenticationPluginManager $manager = NULL) {
+    $manager = $manager ?: AuthenticationPluginManager::create();
```

Comment:
lets still keep this php 5.3 ``$manager ? $manager : AuthenticationPluginManager::create();``